index.js

```diff
@@ -87,6 +107,22 @@ MarketplaceClient.prototype.validatePackage = function(packagePath) {
         }, function(error, response, body) {
           if (error) {
             reject(error);
+          } else if (response.statusCode == 202) {
+            // Checks the validation's status every five seconds, if it's not yet processed
+            var poller = function() {
+              setTimeout(function() {
+                self.getValidation(body.id).then(function(validation) {
+                  if (validation.processed) {
+                    resolve(validation);
+                  } else {
+                    poller();
+                  }
+                }, function() {
```

Comment:
Yea we have an issue opened for those scenarios https://github.com/mozilla/node-firefox-marketplace/issues/4